syntax = "proto3";
package peer;

service PeerService {
    rpc Announce (PeerAddress) returns (PeerAddress) {}
    rpc ElectLeader (ElectionMsg) returns (Response) {}
    rpc ExitNetwork (PeerAddress) returns (Response) {}
    rpc PingBloom (BloomFilter) returns (Response) {}
}

message ElectionMsg {
    bytes id = 1;
}

message Response {
    string msg = 1;
}

message PeerAddress {
    string ip = 1;
    int32 port = 2;
}

message BloomFilter {
    PeerAddress address = 1;
    bytes bloom = 2;
}